### PR TITLE
Add video output option

### LINKED ISSUE #12: Add optional output argument
Would be nice to do `-o /path/to/output` so I don't need a monitor or be SSH'd with X to see the results.

### PR DESCRIPTION
**Pull request by JacksonCoder**
Fixes #12. Specified via `-o` or `--output` option.

### CODE REVIEW

**Comment by JacksonCoder** [comment]
Hey @andrewda, could you please review this when you get the chance?

File: vision/app.py [hunk 1]
```diff
@@ -86,16 +88,23 @@ def run_video(self):
...
             im = camera.read()
             try:
-                lowerThreshold = np.array([nt_utils.get_number("front_lower_blue"), nt_utils.get_number("front_lower_green"), nt_utils.get_number("front_lower_red")])
-                upperThreshold = np.array([nt_utils.get_number("front_upper_blue"), nt_utils.get_number("front_upper_green"), nt_utils.get_number("front_upper_red")])
+                lowerThreshold = np.array(
+                    [nt_utils.get_number("front_lower_blue"), nt_utils.get_number("front_lower_green"),
+                     nt_utils.get_number("front_lower_red")])
+                upperThreshold = np.array(
+                    [nt_utils.get_number("front_upper_blue"), nt_utils.get_number("front_upper_green"),
+                     nt_utils.get_number("front_upper_red")])
```
Comment (andrewda) [inline vision/app.py]: these changes should be done in a new pr

File: vision/app.py [hunk 2]
```diff
@@ -116,7 +125,8 @@ def run_video(self):
                     totalArea = area1 + area2
                     if (totalArea > self.min_area) and (totalArea < self.max_area):
                         if verbose:
-                            print("[Blob] x: %d, y: %d, width: %d, height: %d, total area: %d" % (x1, y1, w1, h1, totalArea))
+                            print("[Blob] x: %d, y: %d, width: %d, height: %d, total area: %d" % (
+                                x1, y1, w1, h1, totalArea))
```
Comment (andrewda) [inline vision/app.py]: new pr

File: vision/app.py [hunk 3]
```diff
@@ -144,7 +154,9 @@ def run_video(self):
 
                     if self.display:
                         cv2.imshow("Original", im)
-
+                # Write to video file
+                if self.output_file:
+                    videoWrite.write(im)
```
Comment (andrewda) [inline vision/app.py]: newline after

File: vision/app.py [hunk 4]
```diff
@@ -86,16 +88,23 @@ def run_video(self):
             print("No image path specified, reading from camera video feed")
 
         timeout = 0
-
```
Comment (andrewda) [inline vision/app.py]: keep newline

**Comment by JacksonCoder** [comment]
Sorry, when I ran auto-style it restyled that stuff without me realizing.

File: vision/app.py [hunk 5]
```diff
@@ -144,6 +150,9 @@ def run_video(self):
 
                     if self.display:
                         cv2.imshow("Original", im)
+                # Write to video file
```
Comment (andrewda) [inline vision/app.py]: newline before

**Comment by JacksonCoder** [comment]
OK, take a look now.

**Comment by JacksonCoder** [comment]
OK, take a look now.

**Comment by SUPERCILEX** [comment]
@JacksonCoder Travis died

**Comment by JacksonCoder** [comment]
Flake8 didn't like the newline I added in the last commit.

File: vision/app.py [hunk 6]
```diff
@@ -144,6 +150,10 @@ def run_video(self):
 
                     if self.display:
                         cv2.imshow("Original", im)
+
```
Comment (andrewda) [inline vision/app.py]: can't have whitespace on an empty line

**Pull request merged**